libarchive/archive_write.3

```diff
@@ -186,8 +186,13 @@ write_archive(const char *outname, const char **filename)
 
   a = archive_write_new();
   mydata->name = outname;
-  archive_write_add_filter_gzip(a);
-  archive_write_set_format_ustar(a);
+  /* Set archive format and filter according to output file extension.
+   * If it fails, set default format. Platform depended function.
+   * See supported formats in archive_write_set_format_filter_by_ext.c */
+  if ( archive_write_set_format_filter_by_ext(a, outname) != ARCHIVE_OK )  {
```

Comment:
Libarchive follows BSD formatting:  there should be no space after '(' or before ')'